strawberry/extensions/__init__.py

```diff
@@ -0,0 +1,92 @@
+import functools
+import typing
+from contextlib import contextmanager
+
+from graphql import MiddlewareManager
+
+
+class Extension:  # pragma: no cover
```

Comment:
Should this be abstract?

Reply:
In terms of extending the abc base class?

Reply:
yeah

Reply:
Probably! Will update the code tomorrow, was going to write some docs for this first, but I'm too tired now ☺️

Reply:
@BryceBeagle I changed the class to be used as baseclass, the empty methods are fine now, they shouldn't be abstract as they are always called :)